src/controller/expensesController.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+
+const { ExpensesService } = require('../services/expenses');
+const { usersController } = require('./usersController');
+
+class ExpensesController extends ExpensesService {
+  constructor() {
+    super();
+
+    this.postExpense = this.postExpense.bind(this);
+    this.createExpense = this.createExpense.bind(this);
+    this.getExpenses = this.getExpenses.bind(this);
+    this.getAll = this.getAll.bind(this);
+    this.getExpense = this.getExpense.bind(this);
+    this.getOne = this.getOne.bind(this);
+    this.removeExpense = this.removeExpense.bind(this);
+    this.removeOne = this.removeOne.bind(this);
+    this.patchExpense = this.patchExpense.bind(this);
+    this.modifyExpence = this.modifyExpence.bind(this);
+  }
+
+  postExpense(req, res) {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    const users = usersController.getAll();
+    const hasUser = users.find(user => +userId === user.id);
+
+    const isValid = hasUser && spentAt && title && amount && category && note;
+
+    const error = hasUser
+      ? 'required parameters is not passed'
+      : 'user does\'nt exist';
+
+    if (!isValid) {
+      res.status(400);
+      res.json({ error });
+
+      return;
+    }
+
+    const expense = super.createExpense({
+      userId, spentAt, title, amount, category, note,
+    });
+
+    res.statusCode = 201;
+    res.json(expense);
+  }
+
+  getExpenses(req, res) {
+    const expenses = super.getAll();
+
+    const { userId, category, to, from } = req.query;
+
+    if (from && to) {
+      const expensesByDate = expenses.filter(
+        (expense) => expense.spentAt > from && expense.spentAt < to
+      );
+
+      res.statusCode = 200;
+      res.json(expensesByDate);
+
+      return;
```

Comment:
You need to filter expenses by multiple options, not only `from` & `to`. Combine all conditions and return filtered array of expenses at once.